ouroboros-consensus-cardano/tools/db-analyser/Analysis.hs

```diff
@@ -200,15 +202,20 @@ storeLedgerStateAt slotNo (AnalysisEnv { db, registry, initLedger, cfg, limit, l
     putStrLn $ "About to store snapshot of a ledger at " <>
                show slotNo <> " " <>
                "this might take a while..."
-    void $ processAll db registry GetBlock limit initLedger process
+    void $ processAllUntil db registry GetBlock limit initLedger process
   where
-    process :: ExtLedgerState blk -> blk -> IO (ExtLedgerState blk)
+    process :: ExtLedgerState blk -> blk -> IO (NextStep, ExtLedgerState blk)
     process oldLedger blk = do
       let ledgerCfg     = ExtLedgerCfg cfg
           appliedResult = tickThenApplyLedgerResult ledgerCfg blk oldLedger
           newLedger     = either (error . show) lrResult $ runExcept $ appliedResult
-      when (slotNo == blockSlot blk) $ storeLedgerState blk newLedger
-      return newLedger
+      when (blockSlot blk == slotNo) $ storeLedgerState blk newLedger
```

Comment:
Consider calling `fail` if we reach `Stop` without ever having seen `blockSlot blk == slotNo`.

This will minimize surprise for a user with a typo in their desired slot number.

Reply:
Are even writing the ledger snapshot for the next available slot and then issuing a WARNING that they specified an empty slot.